pkg/kubectl/service.go

```diff
@@ -196,5 +196,8 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 			return nil, fmt.Errorf("unknown session affinity: %s", params["session-affinity"])
 		}
 	}
+	if params["headless"] == "true" {
```

Comment:
this is based the create-external-load-balancer line:

`if params["create-external-load-balancer"] == "true"`

If we do want to use `kubectl.GetBool`, shouldn't it be taking place in `generate.go` instead?
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate.go#L107

Secondly, what should the behavior be if there is a parsing error?